[Doc] Add `Tensor.Shape`

isdanni opened the pull request:
Summary:
Add `Tensor.Shape` doc.

Fix: #104038


Ref:

- https://github.com/pytorch/pytorch/issues/5544
- https://github.com/pytorch/pytorch/issues/1980

Differential Revision: D47278630

CC: @svekars @carljparker

vadimkantorov commented:
Also, if `tensor.shape` is considered recommended over more ancient `tensor.size()`, it would be nice to have it mentioned explicitly in docs

isdanni commented:
> Also, if `tensor.shape` is considered recommended over more ancient `tensor.size()`, it would be nice to have it mentioned explicitly in docs (and maybe should have `See also: tensor.size()` and mention that it's equivalent?)

Thanks for the review! I added `see also` reference. I do agree `shape` makes more sense intuitively but it's really personal preference picking which to use(As seen in many codebase). I'd advise against including it in the official documentation for now.

isdanni commented:
Can anyone help me review the updated diff? :) 

@vadimkantorov @svekars @carljparker

mikaylagawarecki commented on torch/_tensor_docs.py:
```diff
@@ -4727,6 +4727,27 @@ def callable(a, b) -> number
+    r"""
+shape() -> torch.Size
+
+Returns the size of the :attr:`self` tensor. Tensor :attr:`shape` is the alias for :attr:`size`.
+It was added for consistency with NumPy.
```
```suggestion
Returns the size of the :attr:`self` tensor. Alias for :attr:`size`.
```

mikaylagawarecki reviewed (approved):
Thanks!

isdanni committed:
[Doc] Add `Tensor.Shape` (#104750)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/104750

Add `Tensor.Shape` doc.

Ref:

- https://github.com/pytorch/pytorch/issues/5544
- https://github.com/pytorch/pytorch/issues/1980

Differential Revision: D47278630

fbshipit-source-id: aa2466cba60b21860338e1ea6b8850f4395b6255

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge

mikaylagawarecki commented:
@pytorchbot merge

aallahyar commented:
> Also, if `tensor.shape` is considered recommended over more ancient `tensor.size()`, it would be nice to have it mentioned explicitly in docs (and maybe should have `See also: tensor.size()` and mention that it's equivalent?)



> > Also, if `tensor.shape` is considered recommended over more ancient `tensor.size()`, it would be nice to have it mentioned explicitly in docs (and maybe should have `See also: tensor.size()` and mention that it's equivalent?)
> 
> Thanks for the review! I added `see also` reference. I do agree `shape` makes more sense intuitively but it's really personal preference picking which to use(As seen in many codebase). I'd advise against including it in the official documentation for now.

@isdanni would you also be against adding a "recommendation" statement in the documentation which promotes the use of `shape` in new codes?

I think not only `shape` is more intuitive, it is also more in line with [Numpy's `shape`](https://numpy.org/doc/2.2/reference/generated/numpy.shape.html#numpy.shape).
